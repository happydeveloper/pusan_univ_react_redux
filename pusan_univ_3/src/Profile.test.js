import React from 'react';
import { render } from '@testing-library/react';
import Profile from './Profile';

describe('<Profile />', () => {
  it('matches snapshot', () => {
    const utils = render(<Profile username="duruops" name="강두루" />);
    expect(utils.container).toMatchSnapshot();
  });
  it('shows the props correctly', () => {
    const utils = render(<Profile username="duruops" name="두루" />);
    utils.getByText('duruops'); // duru 라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText('(두루)'); // (두루) 이라는 텍스트를 가진 엘리먼트가 있는지 확인
    utils.getByText(/두/); // 정규식 /두/ 을 통과하는 엘리먼트가 있는지 확인
  });
});