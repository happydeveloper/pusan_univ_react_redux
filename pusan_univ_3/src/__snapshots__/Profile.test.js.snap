// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<div>
  <div>
    <b>
      duruops
    </b>
     
    <span>
      (
      강두루
      )
    </span>
  </div>
</div>
`;
